lib/flowViewport/sceneIndex/fvpWireframeSelectionHighlightSceneIndex.cpp

```diff
@@ -410,13 +330,130 @@ WireframeSelectionHighlightSceneIndex(
 
     auto operation = [this](const SdfPath& primPath, const HdSceneIndexPrim& prim) -> bool {
         if (prim.primType == HdPrimTypeTokens->instancer) {
-            _CreateInstancerHighlightsForInstancer(prim, primPath);
+            _CreateSelectionHighlightsForInstancer(prim, primPath);
+        }
+        else if (prim.primType == HdPrimTypeTokens->mesh) {
+            _CreateSelectionHighlightsForMesh(prim, primPath);
         }
+#if PXR_VERSION >= 2403
+        else if (prim.primType == HdPrimTypeTokens->geomSubset) {
+            _CreateSelectionHighlightsForGeomSubset(primPath);
+        }
+#endif
         return true;
     };
     _ForEachPrimInHierarchy(SdfPath::AbsoluteRootPath(), operation);
 }
 
+// Computes the mask to use for an instancer's selection highlight mirror
+// based on the instancer's topology and its selections.
+VtBoolArray
+WireframeSelectionHighlightSceneIndex::_GetSelectionHighlightMask(const HdInstancerTopologySchema& originalInstancerTopology, const HdSelectionsSchema& selections) const
+{
+    // Schema getters were made const in USD 24.05 (specifically Hydra API version 66).
+    // We work around this for previous versions by const casting.
+    VtBoolArray originalMask = 
+#if HD_API_VERSION < 66
+    const_cast<HdInstancerTopologySchema&>(originalInstancerTopology).GetMask()->GetTypedValue(0);
+#else
+    originalInstancerTopology.GetMask()->GetTypedValue(0);
+#endif
+
+    size_t nbInstances = 0;
+    auto instanceIndices = 
+#if HD_API_VERSION < 66
+    const_cast<HdInstancerTopologySchema&>(originalInstancerTopology).GetInstanceIndices();
+#else
+    originalInstancerTopology.GetInstanceIndices();
+#endif
+    for (size_t iInstanceIndex = 0; iInstanceIndex < instanceIndices.GetNumElements(); iInstanceIndex++) {
+        auto protoInstances = instanceIndices.GetElement(iInstanceIndex)->GetTypedValue(0);
+        nbInstances += protoInstances.size();
+    }
+    if (!TF_VERIFY(originalMask.empty() || originalMask.size() == nbInstances, "Instancer mask has incorrect size.")) {
+        return originalMask;
+    }
+    VtBoolArray selectionHighlightMask = [&]() {
+        if (!selections.IsDefined()) {
+            return originalMask.empty() ? VtBoolArray(nbInstances, true) : originalMask;
+        }
+        return VtBoolArray(nbInstances, false);
+    }();
+
+    if (!selections.IsDefined()) {
```

Comment:
This is the newly handled case and why this function and `_GetSelectionHighlightInstancerDataSource` was moved to a method, in order to call `GetSelectionHighlightPath`